load.py

```diff
@@ -34,15 +34,26 @@
     if os.path.exists(x) and os.path.exists(y):
         full_dataset.append((x, y))
 
+print 'full dataset size : %i' % len(full_dataset)
+
 training_size = int(len(full_dataset) * 0.9)
 val_size = len(full_dataset) - training_size
 train_paths = full_dataset[:training_size]
 val_paths = full_dataset[training_size:]
 print 'num of triaining samples %d' % len(train_paths)
 
+# make sure we don't go off index
+if val_size < VAL_SAMPLES:
+    VAL_SAMPLES = val_size
+if training_size < TRAIN_SAMPLES:
+    TRAIN_SAMPLES = training_size
+
+print 'VAL_SAMPLES: %d' % VAL_SAMPLES
+print 'TRAIN SAMPLES : %d' % TRAIN_SAMPLES
+
 X_train = np.zeros((TRAIN_SAMPLES, 1, 361, 720), dtype=np.float)
 Y_train = np.zeros(TRAIN_SAMPLES, dtype=np.float)
-for i, (x_path, y_path) in enumerate(train_paths[:TRAIN_SAMPLES]):
+for i, (x_path, y_path) in enumerate(train_paths[:TRAIN_SAMPLES-1]):
```

Comment:
OK got it. I misunderstood the [:x] behavior of python.
(still does not explain why it correct the IndexError, will investigate)